waspc/src/Wasp/Generator/ServerGenerator/AuthG.hs

```diff
@@ -105,3 +133,85 @@ genMeRoute auth = return $ C.mkTmplFdWithDstAndData tmplFile dstFile (Just tmplD
+    externalAuthAssociationEntityName = maybe "undefined" AS.refName (AS.Auth.externalAuthAssociationEntity auth)
+    tmplFile = C.srcDirInServerTemplatesDir </> SP.castRel utilsFileInSrcDir
+    dstFile = C.serverSrcDirInServerRootDir </> utilsFileInSrcDir
+    tmplData =
+      object
+        [ "userEntityUpper" .= (userEntityName :: String),
+          "userEntityLower" .= (Util.toLowerFirst userEntityName :: String),
+          "externalAuthAssociationEntityLower" .= (Util.toLowerFirst externalAuthAssociationEntityName :: String),
+          "failureRedirectPath" .= AS.Auth.onAuthFailedRedirectTo auth,
+          "successRedirectPath" .= getOnAuthSucceededRedirectToOrDefault auth
+        ]
+
+    utilsFileInSrcDir :: Path' (Rel C.ServerSrcDir) File'
+    utilsFileInSrcDir = [relfile|routes/auth/utils.js|]
+
+genGoogleAuth :: AS.Auth.Auth -> Generator [FileDraft]
+genGoogleAuth auth
+  | AS.Auth.isGoogleAuthEnabled auth =
+      sequence
+        [ copyTmplFile [relfile|routes/auth/passport/google/google.js|],
+          copyTmplFile [relfile|routes/auth/passport/google/googleDefaults.js|],
+          genGoogleConfigJs auth
+        ]
+  | otherwise = return []
+  where
+    copyTmplFile = return . C.mkSrcTmplFd
+
+genGoogleConfigJs :: AS.Auth.Auth -> Generator FileDraft
+genGoogleConfigJs auth = return $ C.mkTmplFdWithDstAndData tmplFile dstFile (Just tmplData)
+  where
+    tmplFile = C.srcDirInServerTemplatesDir </> SP.castRel googleConfigFileInSrcDir
+    dstFile = C.serverSrcDirInServerRootDir </> googleConfigFileInSrcDir
+    tmplData =
+      object
+        [ "doesConfigFnExist" .= isJust maybeConfigFn,
+          "configFnImportStatement" .= fromMaybe "" maybeConfigFnImportStmt,
+          "configFnIdentifier" .= fromMaybe "" maybeConfigFnImportIdentifier,
+          "doesOnSignInFnExist" .= isJust maybeGetUserFieldsFn,
+          "getUserFieldsFnImportStatement" .= fromMaybe "" maybeOnSignInFnImportStmt,
+          "getUserFieldsFnIdentifier" .= fromMaybe "" maybeOnSignInFnImportIdentifier
+        ]
+
+    googleConfigFileInSrcDir :: Path' (Rel C.ServerSrcDir) File'
+    googleConfigFileInSrcDir = [relfile|routes/auth/passport/google/googleConfig.js|]
+
+    maybeConfigFn = AS.Auth.configFn =<< AS.Auth.google (AS.Auth.methods auth)
+    maybeConfigFnImportDetails = getJsImportDetailsForExtFnImport relPosixPathFromGoogleAuthDirToExtSrcDir <$> maybeConfigFn
+    (maybeConfigFnImportIdentifier, maybeConfigFnImportStmt) = (fst <$> maybeConfigFnImportDetails, snd <$> maybeConfigFnImportDetails)
+
+    maybeGetUserFieldsFn = AS.Auth.getUserFieldsFn =<< AS.Auth.google (AS.Auth.methods auth)
+    maybeOnSignInFnImportDetails = getJsImportDetailsForExtFnImport relPosixPathFromGoogleAuthDirToExtSrcDir <$> maybeGetUserFieldsFn
+    (maybeOnSignInFnImportIdentifier, maybeOnSignInFnImportStmt) = (fst <$> maybeOnSignInFnImportDetails, snd <$> maybeOnSignInFnImportDetails)
+
+-- | TODO: Make this not hardcoded!
```

Comment:
Oh no this TODO is spreading :D!
But all ok, not worth the trouble right now, just found it funny.

Reply:
Yeah, it has been copy-pasted out quite a bit now lol